File: ProductService/internal/dtos/audit_log.go
package dtos

type AuditActor struct {
	ID        string `json:"id"`
	IP        string `json:"ip"`
	UserAgent string `json:"userAgent"`
}

type AuditLog struct {
	Timestamp string                 `json:"timestamp"`
	Actor     AuditActor             `json:"actor"`
	Action    string                 `json:"action"`
	Resource  string                 `json:"resource"`
	Service   string                 `json:"service"`
	Details   map[string]interface{} `json:"details"`
}

File: UserService/UserService/DTOs/Audit/AuditLog.cs
using System.Text.Json.Serialization;

namespace UserService.DTOs.Audit
{
    public class AuditLog
    {
        [JsonPropertyName("timestamp")]
        public string Timestamp { get; set; } = DateTime.UtcNow.ToString("o"); // ISO 8601 format

        [JsonPropertyName("actor")]
        public AuditActor Actor { get; set; }

        [JsonPropertyName("action")]
        public string Action { get; set; }

        [JsonPropertyName("resource")]
        public string Resource { get; set; }

        [JsonPropertyName("service")]
        public string Service { get; set; }

        [JsonPropertyName("details")]
        public Dictionary<string, object> Details { get; set; }
    }
}
